app/models/mixins/retirement_mixin.rb

```diff
@@ -47,6 +47,10 @@ def retirement_warn=(days)
     end
   end
 
+  def retire_request(user)
```

Comment:
@d-m-u Should this also have a group in case a user belongs to multiple groups

Reply:
@mkanoor I think that's a bigger problem and should be addressed in a separate PR.

Reply:
@d-m-u Shouldn't this have parity with the way the retire request gets built from the UI. There is a current_user object that it uses, something akin needs to be used here. Is this a user object or a userid field. When we do provisioning from inside of an automate method

Like here

https://github.com/ManageIQ/manageiq-automation_engine/blob/34ca401a667233c49f4952c78cf4a43883837ac6/lib/miq_automation_engine/engine/miq_ae_method_service/miq_ae_service_methods.rb#L110

Or here

https://github.com/ManageIQ/manageiq-automation_engine/blob/34ca401a667233c49f4952c78cf4a43883837ac6/lib/miq_automation_engine/engine/miq_ae_method_service/miq_ae_service_methods.rb#L115

Maybe the retire request needs to be in the same file for consistency. We already have provisioning request and automation request there.

Reply:
Hi @mkanoor Are you okay if we use the ```User.current_user``` as its done here:
https://github.com/ManageIQ/manageiq-automation_engine/blob/34ca401a667233c49f4952c78cf4a43883837ac6/lib/miq_automation_engine/engine/miq_ae_method_service/miq_ae_service_methods.rb#L116